src/marias3.c

```diff
@@ -205,7 +205,7 @@ ms3_st *ms3_init(const char *s3key, const char *s3secret,
   ms3->disable_verification = false;
   ms3->first_run = true;
   ms3->path_buffer = ms3_cmalloc(sizeof(char) * 1024);
-  ms3->query_buffer = ms3_cmalloc(sizeof(char) * 1024);
+  ms3->query_buffer = ms3_cmalloc(sizeof(char) * 3072);
```

Comment:
I wonder what is the reason to choose such an unusuall size for a buffer? The buffer size is usually is a power of 2. Will it make it worse if we raise it up to 4096 and make it a constant to avoid manual change if we need to alter the value in future?

Reply:
Some new values possible here have a maximum length of 2048 so buffer was increased by that amount.

Reply:
So why shouldn't we make it a constant?